### DISCUSSION TITLE
Switching from standard to eco mode not working on 2023 g14.

### DISCUSSION THREAD

**Discussion by jetthansley** [seerge/g-helper#692] [General]
I already have the setting to "stop all apps from using GPU when switching to eco" but when my laptop has been open for a while and i go to switch to eco, it doesn't do it and just gives me an option to restart my gpu in device manager or something like that, which still doesn't fix the issue. Is there a way to also add the button to stop all gpu processes on the newer models? Also, when freshly rebooted, the switching between standard and eco works just fine.

**Comment by seerge** [comment]
@jetthansley hello, stopping all GPU apps actually does stop them (you can check log with a kill-list at``%appdata%\ghelper`` ). 

Based on what I was reading on reddit, 2023 models sometimes indeed struggle with disabling GPU (according to users using both AC or g-helper). It seem to be some issue in bios implementation atm.

1. AC demands display to be exactly in "Optimus" mode prior to setting Eco. Do you have optimus ON ? (In latest 0.91 version app will also check for that and show a warning if not)

2. And this "restart GPU in device manager" thing is actually a last-resort trick that I added specifically for 2023 models :) But if even that doesn't help, I'm afraid I can't do much on this stage. 

3. How does it work with AC for you? does it set Eco mode all the time? or also only after fresh reboot?

P.S. You may check a topic dedicated to this issue https://github.com/seerge/g-helper/issues/345

**Comment by newKJW** [comment]
1. switch the display mode in the order of ultimate(dgpu only)-standard to initialize the settings.
2. check nvidia control panel, there is 'Manage Display Mode(MDM)'
3. switch to eco-mode
5. switch to standard-mode
6. check nvidia control panel, MDM is missing!
7. switch to eco-mode
8. error occurs(i set MDM to optimus before the option disapeared)
9. i think to-eco-mode only works properly when 'MDM' exists, and i think eco-to-standard works with error(MDM disappears)
10. BTW, to restore MDM, kill every processes by pressing power button for 10 seconds then reboot(MDM restored with the mode you set before. in my case, optimus)
11. if i remember right, AC had same problem

**Reply by seerge** [reply]
@Kojunwon i can give you same advice - is to explicitly set Display Mode to Optimus, prior to disabling GPU :) (that's what Armoury Crate demands as well)

**Reply by newKJW** [reply]
i already set display mode to optimus with control panel before disabling dgpu(eco mode). 'explicity set' is other method?

**Reply by seerge** [reply]
You didn't mention it in your list :) If it was already in "Optimus" it's fine. But as I understand in your case you don't experience any problems with GPU switching itself ? only nvidia control panel loses Display Mode section after re-enabling? Does restarting control panel (or mb any of nvidia services) help? (cause if so - i can try to add that restarting in the app)

**Reply by newKJW** [reply]
after switching eco to standard(no MDM)...
i restarted the dgpu with device manager and waited few seconds and MDM was restored.
after that, i could switch mode standard to eco smoothly.
I finally realized why you made the restart function. thank you
btw, is the 'stop all apps from using GPU when switching to eco' feature for a different problem than the one we just talked about?

**Reply by seerge** [reply]
>btw, is the 'stop all apps from using GPU when switching to eco' feature for a different problem than the one we just talked about?


@Kojunwon initially different, as some apps can hang if GPU is switched (EA Launcher hangs in my case), so I have added this feature, but it's optional as many apps survive gpu power off perfectly (like any browser for example).

But killing all apps can potentially help disabling GPU faster - as it less chances some process will keep it afloat :)

**Reply by newKJW** [reply]
i get it, thx :)

**Comment by seerge** [comment]
@jetthansley @Kojunwon reports that BIOS 310 (available via g-helper / myasus) fixes issue https://github.com/seerge/g-helper/issues/680#issuecomment-1614740415 

Can you check as well?

**Reply by Lucixel10** [reply]
I just updated it to 310 a week ago and i still have that same issue, do you have any fix yet as to what i have to do? TIA

**Reply by seerge** [reply]
@Lucixel10 i don’t think its “fixable” on app level then. As I understand keeping display mode in optimus + killing GPU apps can help sometimes. 

Plus G-Helper would already offer to restart GPU in device manager (as most radical way) if GPU refuses to turn off on first attempt.

Mb try to do a hardware reset, to make sure ?

https://github.com/seerge/g-helper#how-do-i-do-a-hardware-reset-on-a-laptop

**Reply by Lucixel10** [reply]
hi, thanks for the reply. So I actually noticed something. Switching to eco mode from standard mode is possible whenever MDM is active/present as said before. However, shutting down when it's still on eco mode somehow disables MDM the next time the pc is turned on. I guess my fix now is always shut down the pc when on standard mode or just hibernate it so the MDM doesn't disappear.

**Reply by Lucixel10** [reply]
I guess it's because eco mode disables your dGPU and somehow the pc doesnt know how to reactivate it the next time it's on, so you need to turn on standard mode first then shutdown/restart your PC

**Reply by seerge** [reply]
Does somehow restarting nvidia control panel or any of nvidia services help returning MDM ?

**Reply by Lucixel10** [reply]
How do you restart nvidia control panel? If it's simply by exiting and reopening it, it doesn't work.

I'd also like to add what i said. 
When hibernating, do not do it in eco mode. As the next time you switch want to switch from standard mode, you cant switch back to eco mode since that also made MDM disappear.

In other words, do not shutdown or hibernate in eco mode. 

It's annoying and is it fixable with a BIOS update?

**Reply by seerge** [reply]
@Lucixel10 probably fixable (it seem to be waay less complains now after 310 was released), does same happen for you with AC ?

**Reply by seerge** [reply]
@Lucixel10 by the way, do you have Fast boot enabled or disabled in BIOS? try to change it to opposite,

**Reply by Lucixel10** [reply]
Yes before i got this "revelation" AC was the same and doesnt want to switch to eco. Probably the same problem as G-helper as MDM somehow disappeared.

**Reply by Lucixel10** [reply]
I disabled fast boot in power plan. Is it different as in BIOS?

**Reply by seerge** [reply]
@Lucixel10 i'm not sure what kind of fast boot you have in power plan but i mean this https://www.asus.com/support/FAQ/1044641/ :) 

Try to set value to opposite (i.e. disable if it's enabled or vice versa)

**Reply by Lucixel10** [reply]
Yeah sorry i just did that and disabled fast boot in BIOS. The problem remains though. The MDM disappeared again as i hibernated in eco mode. I guess the only fix right now is to not hibernate/shutdown in eco mode

**Reply by seerge** [reply]
May be it's just nvidia drivers acting somehow? Did you try to uninstall all nvidia completely using DDU and then install back ?

**Reply by Lucixel10** [reply]
I guess the other way is to use the restart dgpu option that you made. But if i'm not mistaken didn't you say it's kinda "invasive"? What could possibly happen?

**Reply by Lucixel10** [reply]
Just now a new nvidia update just got released and i did check uninstall the driver option first before installing the new update. Though the problem remains

**Reply by seerge** [reply]
@Lucixel10 i would completely uninstall using DDU, and then do a fresh install :) (if you want ofc)

Disabling / enabling devices in device manager in windows doesn't seem to be a normal thing in windows, that's why that option is being offered by the app only as a last resort :)

**Reply by Lucixel10** [reply]
I just did reinstall using DDU, the problem is still the same

**Reply by seerge** [reply]
@Lucixel10 oke! thanks for testing at least :) then let's hope it's eventually get addressed by bios (and maybe nvidia drivers)

**Reply by Lucixel10** [reply]
Yeah hopefully so, keep up the good work to you too :)

**Reply by seerge** [reply]
You can restart a GPU in device manager btw using following powershell command (needs to be run as admin)

```
$device = Get-PnpDevice | Where-Object { $_.FriendlyName -imatch 'NVIDIA' -and $_.Class -eq 'Display' }; Disable-PnpDevice $device.InstanceId -Confirm:$false; Start-Sleep -Seconds 3; Enable-PnpDevice $device.InstanceId -Confirm:$false
```

**Reply by Lucixel10** [reply]
Allright i'll keep that in mind

**Comment by joe-700** [comment]
This has been addressed in version 0.101.0.0. I think the 2023 models suffer from odd behaviour when booting into eco mode. The revision will set standard mode on shutdown to ensure booting into a safe state and then reselect the last user select mode on startup. Hopefully ASUS will address this with a BIOS update in the future but for the time being this works seamlessly.
